src/pages/home/report/ReportActionsList.js

```diff
@@ -171,44 +173,50 @@ function ReportActionsList(props) {
 
     return (
         <Animated.View style={[animatedStyles, styles.flex1]}>
-            <InvertedFlatList
-                accessibilityLabel={props.translate('sidebarScreen.listOfChatMessages')}
-                ref={reportScrollManager.ref}
-                data={props.sortedReportActions}
-                renderItem={renderItem}
-                contentContainerStyle={[styles.chatContentScrollView, shouldShowReportRecipientLocalTime || shouldOmitBottomSpace ? styles.pt0 : {}]}
-                keyExtractor={keyExtractor}
-                initialRowHeight={32}
-                initialNumToRender={calculateInitialNumToRender()}
-                onEndReached={props.loadMoreChats}
-                onEndReachedThreshold={0.75}
-                ListFooterComponent={() => {
-                    if (props.report.isLoadingMoreReportActions) {
-                        return <ReportActionsSkeletonView containerHeight={CONST.CHAT_SKELETON_VIEW.AVERAGE_ROW_HEIGHT * 3} />;
-                    }
-
-                    // Make sure the oldest report action loaded is not the first. This is so we do not show the
-                    // skeleton view above the created action in a newly generated optimistic chat or one with not
-                    // that many comments.
-                    const lastReportAction = _.last(props.sortedReportActions) || {};
-                    if (props.report.isLoadingReportActions && lastReportAction.actionName !== CONST.REPORT.ACTIONS.TYPE.CREATED) {
-                        return (
-                            <ReportActionsSkeletonView
-                                containerHeight={skeletonViewHeight}
-                                animate={!props.network.isOffline}
-                            />
-                        );
-                    }
-                    return null;
-                }}
-                keyboardShouldPersistTaps="handled"
-                onLayout={(event) => {
-                    setSkeletonViewHeight(event.nativeEvent.layout.height);
-                    props.onLayout(event);
-                }}
-                onScroll={props.onScroll}
-                extraData={extraData}
-            />
+            <ReportActionListFrozenScrollContext.Provider value={{setShouldFreezeScroll}}>
```

Comment:
I guess we should wrap entire **ReportScreen**. I believe it could help in the future to get the value and setter somewhere within the screen.

Reply:
In general, the approach is to create a [HOC](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#composition-vs-inheritance), which isn't clean to throw the Provider in the middle of a component. 
Also if possible, given that it is a function component, maybe a hook would work.

I would simply wrap this component with the hook, I wouldn't add it at the `ReportScreen` level, given that if not needed, any change will rerender all the children's components